gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/cockpit/command/handler/GoodbyeCommandHandlerTest.java

```diff
@@ -91,10 +106,64 @@ public void handleWithException() {
 
         when(installationService.getOrInitialize()).thenReturn(installation);
         when(installationService.setAdditionalInformation(anyMap())).thenThrow(new TechnicalManagementException());
+        when(promotionService.search(any(), any(), any())).thenReturn(new Page<>(emptyList(), 0, 0, 0));
 
         TestObserver<GoodbyeReply> obs = cut.handle(command).test();
 
         obs.awaitTerminalEvent();
         obs.assertValue(reply -> reply.getCommandId().equals(command.getId()) && reply.getCommandStatus().equals(CommandStatus.ERROR));
     }
+
+    @Test
+    public void handleRejectAllPromotionToValidate() {
+        final InstallationEntity installation = new InstallationEntity();
+        installation.setId(INSTALLATION_ID);
+        installation.getAdditionalInformation().put(CUSTOM_KEY, CUSTOM_VALUE);
+
+        GoodbyeCommand command = new GoodbyeCommand();
+
+        when(installationService.getOrInitialize()).thenReturn(installation);
+
+        PromotionEntity promotionEntity1 = getAPromotionEntity("promotion#1");
+        PromotionEntity promotionEntity2 = getAPromotionEntity("promotion#2");
+        when(promotionService.search(any(), any(), any())).thenReturn(new Page<>(List.of(promotionEntity1, promotionEntity2), 0, 0, 0));
+
+        TestObserver<GoodbyeReply> obs = cut.handle(command).test();
+
+        obs.awaitTerminalEvent();
+        obs.assertValue(reply -> reply.getCommandId().equals(command.getId()) && reply.getCommandStatus().equals(CommandStatus.SUCCEEDED));
+
+        ArgumentCaptor<PromotionEntity> captor = ArgumentCaptor.forClass(PromotionEntity.class);
+
+        verify(promotionService, times(2)).createOrUpdate(captor.capture());
+
+        assertThat(captor.getAllValues())
+            .extracting(PromotionEntity::getId, PromotionEntity::getStatus)
+            .containsExactlyInAnyOrder(
+                tuple(promotionEntity1.getId(), PromotionEntityStatus.REJECTED),
+                tuple(promotionEntity2.getId(), PromotionEntityStatus.REJECTED)
+            );
+    }
+
+    private PromotionEntity getAPromotionEntity(String id) {
```

Comment:
@gravitee-io/apim we should think to a way to mutualize all this test utility functions somewhere

Reply:
We should create a `faker` dedicated package, wdyt ?

Reply:
I will do it in another PR ;)